### PR TITLE
Add drift command to display resources that have been changed outside of CFN

### PR DESCRIPTION
**Pull request by stevehodgkiss**
Uses the CFN drift detection API to trigger drift detection & display drifted resources:

stackmaster drift feature

### CODE REVIEW

File: lib/stack_master/cli.rb [hunk 1]
```diff
@@ -215,6 +215,17 @@ def execute!
         end
       end
 
+      command :drift do |c|
+        c.syntax = 'stack_master drift [region_or_alias] [stack_name]'
+        c.summary = 'Detects and displays stack drift'
```
Comment (jacobbednarz) [inline lib/stack_master/cli.rb:220]: Is it assumed here that "displays stack drift" means per Cloudformation standards and not what stack master has?

Comment (joesustaric) [reply lib/stack_master/cli.rb:220]: Yup I have similar question, I have not looked at the code but I assume this is a drift between the current aws stack definition vs the actual stack resources and config. (i.e not the stack master code)

P.S - big +1 to including this stuff 🏆

Comment (jacobbednarz) [reply lib/stack_master/cli.rb:220]: I assumed as much too but do wonder whether other users of this are on the same page.

Comment (petervandoros) [reply lib/stack_master/cli.rb:220]: How about tweaking the summary to be more explicit?
```suggestion
        c.summary = 'Detects and displays stack drift using the CloudFormation Drift API'
```

Comment (patrobinson) [reply lib/stack_master/cli.rb:220]: Yep I think we need documentation to clarify to users clearly what the difference between `diff` and `drift` is, as it's easily confused.

File: lib/stack_master/commands/drift.rb [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+      private
+
+      def cf
+        @cf ||= StackMaster.cloud_formation_driver
+      end
+
+      def display_drift(drift)
+        color = drift_color(drift)
+        puts colorize("#{drift.stack_resource_drift_status} #{drift.resource_type} #{drift.logical_resource_id} #{drift.physical_resource_id}", color)
```
Comment (patrobinson) [inline lib/stack_master/commands/drift.rb:37]: I rarely care about line length but it's making it a little hard to review this PR.

File: lib/stack_master/commands/drift.rb [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+      end
+
+      def stack_drift_status_color(stack_drift_status)
+        case stack_drift_status
+        when 'IN_SYNC'
+          :green
+        when 'DRIFTED'
+          :yellow
+        else
+          :blue
```
Comment (patrobinson) [inline lib/stack_master/commands/drift.rb:70]: What other states are there?

File: lib/stack_master.rb [hunk 4]
```diff
@@ -198,4 +199,28 @@ def stderr
...
+    diff.each_line do |line|
+      if line.start_with?('+')
+        stdout.print colorize(line, :green)
+      elsif line.start_with?('-')
+        stdout.print colorize(line, :red)
+      else
+        stdout.print line
+      end
+    end
+  end
```
Comment (patrobinson) [inline lib/stack_master.rb:225]: It feels like this deserves it's own class

File: lib/stack_master/commands/drift.rb (hunk 3 above)
Comment (stevehodgkiss) [reply lib/stack_master/commands/drift.rb:70]: > String, one of "DRIFTED", "IN_SYNC", "UNKNOWN", "NOT_CHECKED"

**Comment by stevehodgkiss** [comment]
> I wonder if we should fail if any of the states are not as expected? I think I prefer to just bail if it's not IN_SYNC/DRIFTED.

If there are no resource drifts to display in the second part (`cf.describe_stack_resource_drifts`), it will exit without displaying anything. I'm not sure how you end up in the `UNKNOWN` state, but there might information in the resource drifts to display, so I think it's reasonable to still attempt it 🤷 

> Other than that I would love to see some good practical examples of what the difference between drift and diff is in the README.

I've added some info about the 2 commands to the readme. I might update the readme with example screenshots after this has been merged and released. The apply demo is from before changesets were implemented, so it could do with an update.

File: lib/stack_master/commands/drift.rb [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+      def display_resource_drift(drift)
+        diff = ::StackMaster::Diff.new(before: prettify_json(drift.expected_properties),
+                        after: prettify_json(drift.actual_properties))
+        diff.display_colorized_diff
+      end
+
+      def prettify_json(string)
+        JSON.pretty_generate(JSON.parse(string)) + "\n"
+      rescue => e
```
Comment (orien) [inline lib/stack_master/commands/drift.rb:61]: Nitpick: let's rescue a specific error:

```suggestion
      rescue => StandardError e
```

**Comment by stevehodgkiss** [comment]
> Do you want to slap the frozen string literals comment on new files? (it's going to be enforced in ruby 3, so it will be needed anyway)

Not as part of this PR. I'd rather do all files in one sweep in a separate PR.

**Comment by stevehodgkiss** [comment]
> Do you think it's worth a writing a feature test?

I've added an aruba integration test using rspec instead of cucumber. It uses the built in aws client library stubbing mechanism rather than the custom driver class used in cucumber. I think that style could be easier to read and maintain going forward.

**Pull request merged**